src/Microsoft.VisualStudio.Services.Agent/HostContext.cs

```diff
@@ -68,6 +72,19 @@ public HostContext(string hostType, string logFile = null)
             }
 
             _trace = GetTrace(nameof(HostContext));
+            
+            // Enable Http trace
+            bool enableHttpTrace;
+            if (bool.TryParse(Environment.GetEnvironmentVariable("VSTS_AGENT_HTTPTRACE"), out enableHttpTrace) && enableHttpTrace)
```

Comment:
have you considered using a separate trace object, e.g. one named "HttpTrace" or something different than "HostContext"